vlasovsolver/cpu_trans_map_amr.cpp

```diff
@@ -884,13 +902,12 @@ bool checkPencils(const dccrg::Dccrg<SpatialCell,dccrg::Cartesian_Geometry>& mpi
    }
 
    for (int ipencil = 0; ipencil < pencils.N; ++ipencil) {
```

Comment:
This loop goes over every pencil, and for each pencil takes every cell, and for each of those, checks over all id's within all pencils. Just for a sanity check. Sounds super-wasteful!

Plus, we actually already checked the count for each cell separately above.

Would it work to:
    for (auto id : cells) {
      if (mpiGrid[id]->sysBoundaryFlag == sysboundarytype::NOT_SYSBOUNDARY )  {
         int myCount = std::count(pencils.ids.begin(), pencils.ids.end(), id);
 use std::find to get first pencil in which this cell is used
 then use cint nPencilsThroughThisCell = pow(pow(2,pencils.path[ipencil].size()),2); on that pencil and perform the check

Of course this is a valid question only if this takes a non-negligible time to calculate, but it does seem like a fair bit of looping.

Reply:
Also, this is called in a non-threaded section and could surely be threaded

Reply:
The whole "buildPencils" group which includes the pencil-building and this check takes 0.5% of translation. Do we really want to fiddle here?

Reply:
perhaps not.